### PR TITLE
staking analytics for moonbeam

### PR DESCRIPTION
**Pull request by bnsports**

### CODE REVIEW

File: fearless/Common/DataProvider/Sources/Rewards/SubqueryRewardsSource.swift [hunk 1]
```diff
@@ -24,7 +24,117 @@ final class SubqueryRewardsSource {
...
+                result.append("greaterThanOrEqualTo:\"\(timestamp)\",")
+            }
+            if let timestamp = endTimestamp {
+                result.append("lessThanOrEqualTo:\"\(timestamp)\",")
+            }
+            result.append("}")
+            return result
+        }()
+
+        return """
```
Comment (bnsports) [inline fearless/Common/DataProvider/Sources/Rewards/SubqueryRewardsSource.swift:89]: add timestamp filter

File: fearless/Common/Network/Subquery/SubqueryRewardOperationFactory.swift [hunk 2]
```diff
@@ -87,22 +93,67 @@ final class SubqueryRewardOperationFactory {
...
-                     reward: { isNull: false },
-                    \(timestampFilter)
-                 }
-             ) {
-                nodes {
-                    id
-                    timestamp
-                    address
-                    reward
+                    delegators(
```
Comment (bnsports) [inline fearless/Common/Network/Subquery/SubqueryRewardOperationFactory.swift:96]: timestamp filter

File: fearless/Modules/Staking/Analytics/Validators/AnalyticsValidatorsPresenter.swift [hunk 3]
```diff
@@ -74,15 +74,15 @@ extension AnalyticsValidatorsPresenter: AnalyticsValidatorsPresenterProtocol {
...
-        // TODO: Transition with new parameters
-//        wireframe.showValidatorInfo(
-//            chain: chain,
-//            asset: asset,
-//            selectedAccount: selectedAccount,
-//            address: validatorAddress,
-//            view: view
-//        )
+    func handleValidatorInfoAction(validatorAddress: AccountAddress) {
+        // TODO: Parachain case if needed
```
Comment (bnsports) [inline fearless/Modules/Staking/Analytics/Validators/AnalyticsValidatorsPresenter.swift:78]: remove todo

File: fearless/Modules/Staking/StakingMain/StakingMainInteractor+InputProtocol.swift [hunk 4]
```diff
@@ -51,6 +51,11 @@ extension StakingMainInteractor: StakingMainInteractorInputProtocol {
                 for: chainId,
                 accountId: accountId
             )
+
+            if chainAsset.stakingType == .paraChain,
```
Comment (bnsports) [inline fearless/Modules/Staking/StakingMain/StakingMainInteractor+InputProtocol.swift:55]: not beautiful

File: fearless/Modules/Staking/StakingMain/StakingMainInteractor+Subscription.swift [hunk 5]
```diff
@@ -147,15 +147,15 @@ extension StakingMainInteractor {
...
-//                period: .week
-//            )
-//        }
+    func subscribeRewardsAnalytics(for stash: AccountAddress) {
+        if let analyticsURL = selectedChainAsset?.chain.externalApi?.staking?.url {
+            rewardAnalyticsProvider = subscribeWeaklyRewardAnalytics(for: stash, url: analyticsURL)
+        } else {
+            presenter?.didReceieve(
+                subqueryRewards: .success(nil),
+                period: .month
```
Comment (bnsports) [inline fearless/Modules/Staking/StakingMain/StakingMainInteractor+Subscription.swift:156]: week

File: fearless/Modules/Staking/StakingMain/StakingMainInteractor+Subscription.swift [hunk 6]
```diff
@@ -309,10 +309,10 @@ extension StakingMainInteractor: StakingAnalyticsLocalStorageSubscriber,
         address: AccountAddress,
         url _: URL
     ) {
-        guard selectedAccount?.toAddress() == address else {
+        guard selectedAccount?.toAddress()?.lowercased() == address.lowercased() else {
             return
         }
 
-        presenter?.didReceieve(subqueryRewards: result, period: .week)
+        presenter?.didReceieve(subqueryRewards: result, period: .month)
```
Comment (bnsports) [inline fearless/Modules/Staking/StakingMain/StakingMainInteractor+Subscription.swift:316]: week

File: fearless/Modules/Staking/StakingMain/StakingMainViewFactory.swift [hunk 7]
```diff
@@ -272,7 +272,7 @@ final class StakingMainViewFactory: StakingMainViewFactoryProtocol {
             logger: Logger.shared
         )
 
-        let stakingAnalyticsLocalSubscriptionFactory = StakingAnalyticsLocalSubscriptionFactory(
+        let stakingAnalyticsLocalSubscriptionFactory = ParachainAnalyticsLocalSubscriptionFactory(
```
Comment (bnsports) [inline fearless/Modules/Staking/StakingMain/StakingMainViewFactory.swift:275]: add condition to decide what factory we should use

File: fearless/Modules/Staking/StakingMain/View/RewardAnalyticsWidgetView.swift [hunk 8]
```diff
@@ -144,7 +144,7 @@ final class RewardAnalyticsWidgetView: UIView {
 
             applyViewModel()
         } else {
-            startLoading()
+//            startLoading()
```
Comment (bnsports) [inline fearless/Modules/Staking/StakingMain/View/RewardAnalyticsWidgetView.swift:147]: check this

File: fearless/Common/Network/Subquery/SubqueryRewardOperationFactory.swift [hunk 9]
```diff
@@ -220,12 +271,57 @@ extension SubqueryRewardOperationFactory: SubqueryRewardOperationFactoryProtocol
         return operation
     }
 
+    func createDelegatorRewardsOperation(
+        address: String,
```
Comment (DRadmir) [inline fearless/Common/Network/Subquery/SubqueryRewardOperationFactory.swift:275]: use one code style for this file please. You used in 121 line "_ address", but here "address"

File: fearless/Modules/Staking/Analytics/Rewards/Flow/Parachain/AnalyticRewardsParachainViewModelFactory.swift [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+    ) -> AnalyticsSelectedChartData {
+        guard let rewardsByDate = rewardsByDate else {
+            return AnalyticsSelectedChartData(
+                yValue: 0,
+                dateTitle: dateFormatter.string(from: defaultDate),
+                sections: []
+            )
+        }
+
+        let yValue = rewardsByDate.1
```
Comment (DRadmir) [inline fearless/Modules/Staking/Analytics/Rewards/Flow/Parachain/AnalyticRewardsParachainViewModelFactory.swift:134]: The name of the variable is a little misleading, in fact there "yValues"

File: fearless/Modules/Staking/StakingMain/StakingMainInteractor+Subscription.swift [hunk 11]
```diff
@@ -36,7 +36,7 @@ extension StakingMainInteractor {
             }
 
             subscribeToControllerAccount(address: stashItem.controller, chain: chainAsset.chain)
-            subscribeRewardsAnalytics(for: stashItem.stash)
+//            subscribeRewardsAnalytics(for: stashItem.stash)
```
Comment (DRadmir) [inline fearless/Modules/Staking/StakingMain/StakingMainInteractor+Subscription.swift:39]: check this comment plz

File: fearless/Modules/Staking/StakingMain/StakingMainViewController.xib [hunk 12]
```diff
@@ -129,11 +129,6 @@
             <point key="canvasLocation" x="-108.69565217391305" y="111.83035714285714"/>
         </view>
     </objects>
-    <designables>
```
Comment (DRadmir) [inline fearless/Modules/Staking/StakingMain/StakingMainViewController.xib:132]: this is okey?

**Review by DRadmir — CHANGES_REQUESTED** [review]

**Review by DRadmir — APPROVED** [review]

**Pull request merged**